pkg/graveler/retention/garbage_collection_manager_test.go

```diff
@@ -0,0 +1,88 @@
+package retention_test
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"github.com/treeverse/lakefs/pkg/block"
+	"github.com/treeverse/lakefs/pkg/block/mem"
+	"github.com/treeverse/lakefs/pkg/graveler"
+	"github.com/treeverse/lakefs/pkg/graveler/retention"
+	"github.com/treeverse/lakefs/pkg/graveler/testutil"
+)
+
+func TestGarbageCollectionManager_GetUncommittedLocation(t *testing.T) {
+	blockAdapter := mem.New()
+	refMgr := &testutil.RefsFake{}
+	prefix := "test_prefix"
+	runID := "my_test_runID"
+	namespace := "test-namespace"
+	repo := "my-repo"
+	ns := graveler.StorageNamespace(fmt.Sprintf("%s://%s/%s", block.BlockstoreTypeMem, namespace, repo))
+	path := fmt.Sprintf("%s/%s/retention/gc/%s/uncommitted/", ns, prefix, runID)
+	gc := retention.NewGarbageCollectionManager(blockAdapter, refMgr, prefix)
+	location, err := gc.GetUncommittedLocation(runID, ns)
+	require.NoError(t, err)
+	require.Equal(t, path, location)
+}
+
+func createTestFile(t *testing.T, filename, testLine string, count int) {
+	t.Helper()
+	fd, err := os.Create(filename)
+	require.NoError(t, err)
+	defer fd.Close()
+	for i := 0; i < count; i++ {
+		_, err := fd.WriteString(fmt.Sprintf("%s_%d\n", testLine, i))
+		require.NoError(t, err)
+	}
+}
+
+func TestGarbageCollectionManager_SaveGarbageCollectionUncommitted(t *testing.T) {
```

Comment:
Again, I feel that "SaveGarbageCollectionUncommitted" is out of place. It doesn't really do anything related to GC, and this test proves it - you basically test that it saves a file under a given location.